### PR TITLE
Implement configurable line width for debug lines

### PR DESCRIPTION
**Pull request by NCrashed**
Problem: debug lines are too thin for orthographic projection with large top/bottom/left/right ranges (usually used for 2D setups).

Solution:
* Use uniform for line width.
* Pass line width via resource.

Also added the usage of the new feature to example.

How to test:
* Change `DebugLinesParams` in `debug_lines` example and run

### CODE REVIEW

File: examples/debug_lines/main.rs [hunk 1]
```diff
@@ -2,19 +2,16 @@
...
-        cgmath::{Deg, Point3, Vector3},
-        transform::{GlobalTransform, Transform, TransformBundle},
-    },
-    ecs::{Read, System, Write},
-    input::InputBundle,
-    prelude::*,
-    renderer::*,
-    utils::application_root_dir,
-};
+use amethyst::controls::{FlyControlBundle, FlyControlTag};
```
Comment (magnonellie) [inline examples/debug_lines/main.rs]: Did rustfmt do this? It was literally changed a few days ago to the format you removed :stuck_out_tongue:

File: docs/CHANGELOG.md [hunk 2]
```diff
@@ -30,7 +30,7 @@ The format is based on [Keep a Changelog][kc], and this project adheres to
 * Support for multiline text. ([#965])
 * Support for logging to file, toggle for logging to stdout. ([#976], [#994])
 * Added `shadow_update()` and `shadow_fixed_update()` to the `State` trait. ([#1006])
-
+* Added configurable width for debug lines ([#1016])
```
Comment (magnonellie) [inline docs/CHANGELOG.md]: You need to also add a link definition below

File: examples/debug_lines/main.rs (hunk 1 above)
Comment (NCrashed) [reply examples/debug_lines/main.rs]: Yep, it is rustmft `rustfmt 0.99.1-stable (da17b68 2018-08-04)`

Comment (magnonellie) [reply examples/debug_lines/main.rs]: Well then. I guess @jojolepro was right after all. :sweat_smile:

Comment (NCrashed) [reply examples/debug_lines/main.rs]: So, I can revert the particular changes.

Comment (torkleyy) [reply examples/debug_lines/main.rs]: Why are they doing this?

File: docs/CHANGELOG.md (hunk 2 above)
Comment (NCrashed) [reply docs/CHANGELOG.md]: Done

**Comment by NCrashed** [comment]
There is a problem that all lines from components and resource are merged in single mesh. So, all lines are forced to have the same width now. Possible solution is construct several meshes, one per width value.

**Comment by jojolepro** [comment]
Ah that makes sense, you can keep this as-is for now until the new renderer comes and we can have batching (to not kill fps while using debug lines in the mean time)

**Comment by Rhuagh** [comment]
Bors r+

**Pull request merged**